### PR TITLE
Annotate nodes with their Kubernetes components hash

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

- The first bootstrapper labels itself
- The joinservice adds each node as JoiningNode CRD with node name and components hash once it issues a join ticket
- Node Operator consumes the JoiningNode CRD and attaches the label once it sees the node in K8s

Updatability:

- Since the JoinService must create a JoiningNode CRD this must be registered before.
- Operator can be updated independently
- Also note the permission changes needed for the Operator and JoinService

### Checklist

- [x] Link to Milestone

### CODE REVIEW

File: operators/constellation-node-operator/controllers/joiningnode_controller.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		logr.Error(err, "unable to delete JoiningNode")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *JoiningNodesReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// index pending nodes by nodename.
```
Comment (malt3) [inline operators/constellation-node-operator/controllers/joiningnode_controller.go:97]:
```suggestion
	// index joining nodes by nodename.
```

File: joinservice/internal/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -70,6 +80,30 @@ func (c *Client) CreateConfigMap(ctx context.Context, configMap corev1.ConfigMap
...
+		},
+	})
+
+	_, err := c.dynClient.Resource(joiningNodeResource).Create(ctx, joiningNode, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create joining node: %w", err)
+	}
+	return nil
+}
+
```
Comment (malt3) [inline joinservice/internal/kubernetes/kubernetes.go:106]: I think the logic here should be adapted to update the resource if it already exists to ensure we don't get stuck here (could happen if we accidentally issue tickets twice).

**Review by malt3 — COMMENTED** [review]
Some minor changes are needed

**Review by malt3 — APPROVED** [review]
LGTM

**Comment by derpsteb** [comment]
[GCP](https://github.com/edgelesssys/constellation/actions/runs/3646068397) testrun with 2+3 nodes.
[AWS](https://github.com/edgelesssys/constellation/actions/runs/3646075658) testrun with 2+3 nodes.

**Review by derpsteb — APPROVED** [review]

**Pull request merged**